src/components/AppProvider/README.md

```diff
@@ -430,13 +430,48 @@ function AppProviderWithAllThemeKeysExample() {
 
 Translations are provided in the locales folder. When using Polaris, you are able to import translations from all languages supported by the core Shopify product and consume them through the `i18n` prop.
 
+At its simplest you can pass the json content from a locale file into `AppProvider`. This approach is suitable when a project has only one specific locale.
+
 ```jsx
+import AppProvider from '@shopify/polaris';
 // en.json is English. Replace with fr.json for French, etc
 import translations from '@shopify/polaris/locales/en.json';
 
-ReactDOM.render(
-  <AppProvider i18n={translations}>{/* App content */}</AppProvider>,
-);
+function App() {
+  return <AppProvider i18n={translations}>{/* App content */}</AppProvider>;
+}
+```
+
+Alternatively if your project supports multiple locales you can load them dynamically using [@shopify/react-i18n](https://github.com/Shopify/quilt/tree/master/packages/react-i18n#translation). This will ensure you only load the translations you need instead of potentially loading every set of translations even if you are not using them.
+
+```jsx
+import AppProvider from '@shopify/polaris';
+// en.json is English. Replace with fr.json for French, etc
+import translations from '@shopify/polaris/locales/en.json';
+import {useI18n} from '@shopify/react-i18n';
+
+function App() {
+  const [i18n] = useI18n({
+    id: 'Polaris',
+    fallback: translations,
+    translations(locale) {
+      return import(
+        /* webpackChunkName: "Polaris-i18n-[request]", webpackMode: "lazy-once" */ `@shopify/polaris/locales/${locale}.json`
+      ).then((dictionary) => dictionary && dictionary.default);
+    },
+  });
+
+  // i18n.translations is an array of translation dictionaries, where the first
```

Comment:
`translation dictionaries` is a named for a type we've used but consumers may not know what we're referring too. What do you think about describing it the type?

Reply:
That feels like an implementation detail of react-i18n, I don't think we need to go into much detail other than "here's what useI18n gives us, here's how we need to tweak it so it works with our internationalisation".

Reply:
iirc i18n.translations can include various translations. If we're wanting people to use `react-i18n`, they'll want to use it for their custom translations as well rather than have two i18n dependencies. i18n.translations could be [polaris_fallback, polaris_jp, custom translations]. I'm not sure how much detail we'll want to include, but it might help the consumer become more aware of what's actually happening.

Reply:
I don't think it's our responsibility to describe how react-i18n works here - that's what their docs are for. This is true for the common case that we recommend and if people want to use a more advanced setup then it's up to them to work out how they want to diverge from our recommendations.